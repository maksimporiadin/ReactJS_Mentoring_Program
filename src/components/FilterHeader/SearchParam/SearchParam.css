.searchParam {
    padding: 15px 0;
    color: white;
    display: flex;
    flex-flow: row;
    align-items: baseline;
    align-content: space-between;
}

.searchParam-title {
    color: white;
    text-transform: uppercase;
    font-size: 12px;
    padding: 0 20px;
}

.searchParam-input {
    padding: 0 20px;
    margin-right: 20px;
}

.searchParam-input input {
    display: none;
}

.searchParam-grow {
    flex-grow:1;
    text-align:center
}

