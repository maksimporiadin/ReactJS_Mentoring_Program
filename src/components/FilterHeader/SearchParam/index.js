import React, { Component } from 'react';

import { SEARCH_BY } from '../../../App.constants';

import './style.css';

const SearchParam = ({ searchBy, onChange, children}) => (
    <div className="searchParam" onChange={onChange}>
            <span className="searchParam-title">Search By</span>
            <label htmlFor="search-title"
                   className={`btn ${searchBy === SEARCH_BY.TITLE.value? 'btn-primary' : 'btn-secondary'} btn-sm searchParam-input`}>
                <input type="radio"
                       name="search"
                       id="search-title"
                       value={SEARCH_BY.TITLE.value}/> {SEARCH_BY.TITLE.name}
            </label>
            <label htmlFor="search-genre"
                   className={`btn ${searchBy === SEARCH_BY.GENRE.value? 'btn-primary' : 'btn-secondary'} btn-sm searchParam-input`}>
                <input type="radio"
                       name="search"
                       id="search-genre"
                       value={SEARCH_BY.GENRE.value}/> {SEARCH_BY.GENRE.name}
            </label>
        <div className="searchParam-grow"></div>
        {children}
    </div>
)

export default SearchParam;