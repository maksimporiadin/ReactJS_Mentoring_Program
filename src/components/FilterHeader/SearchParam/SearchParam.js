import React, { Component } from 'react';

import { SEARCH_BY } from '../../../App.constants';

import './SearchParam.css';

const SearchParam = ({ searchBy, onChange, children}) => (
    <div className="searchParam" onChange={onChange}>
        <span className="searchParam-title">Search By</span>
        <label htmlFor="search-title"
               className={`btn ${searchBy === SEARCH_BY.TITLE? 'btn-primary' : 'btn-secondary'} btn-sm searchParam-input`}>
            <input type="radio"
                   name="search"
                   id="search-title"
                   value={SEARCH_BY.TITLE}
                   defaultChecked={searchBy === SEARCH_BY.TITLE}/> {SEARCH_BY.TITLE}
        </label>
        <label htmlFor="search-genre"
               className={`btn ${searchBy === SEARCH_BY.GENRE? 'btn-primary' : 'btn-secondary'} btn-sm searchParam-input`}>
            <input type="radio"
                   name="search"
                   id="search-genre"
                   value={SEARCH_BY.GENRE}
                   defaultChecked={searchBy === SEARCH_BY.GENRE}/> {SEARCH_BY.GENRE}
        </label>
        <div className="searchParam-grow"></div>
        {children}
    </div>
)

export default SearchParam;