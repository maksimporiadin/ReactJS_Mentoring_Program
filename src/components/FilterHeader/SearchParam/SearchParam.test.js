import 'jsdom-global/register';
import React from 'react';

import SearchParam from './index';

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('SearchParam', () => {
    let wrapper;

    it('is renders', () => {
        wrapper = shallow(<SearchParam  />)
        expect(wrapper).toHaveLength(1);
    });
});