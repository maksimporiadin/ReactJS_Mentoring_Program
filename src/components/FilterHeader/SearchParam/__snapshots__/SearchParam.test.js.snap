// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchParam is renders 1`] = `
<div
  className="searchParam"
  onChange={[Function]}
>
  <span
    className="searchParam-title"
  >
    Search By
  </span>
  <label
    className="btn btn-secondary btn-sm searchParam-input"
    htmlFor="search-title"
  >
    <input
      defaultChecked={false}
      id="search-title"
      name="search"
      type="radio"
      value="title"
    />
     
    Title
  </label>
  <label
    className="btn btn-secondary btn-sm searchParam-input"
    htmlFor="search-genre"
  >
    <input
      defaultChecked={false}
      id="search-genre"
      name="search"
      type="radio"
      value="genre"
    />
     
    Genre
  </label>
  <div
    className="searchParam-grow"
  />
</div>
`;
